### DISCUSSION TITLE
Sometimes cached files do not appear

### DISCUSSION THREAD

**Discussion by PhilipDeegan** [ccache/ccache#198] [Support]
Hi there,

I'm using ccache with a custom tool that uses pthreads/forks/execvp - and I sometimes see that object files do not appear as output of the ccache gcc -c command

It's quite likely this is an issue in my own code, however I wanted to see if this is something that has been heard of or is known.

Thanks

**Comment by jrosdahl** [comment]
Nothing I've heard of before.

I'm afraid you have to debug the issue yourself to get somewhere. Perhaps by using strace?

**Comment by PhilipDeegan** [comment]
Thanks will give it a go

**Comment by PhilipDeegan** [comment]
It looks like a file descriptor issue as I'm combing threads/forks - silent failure is disconcerting
